### ISSUE TITLE
4x4 mixer mteters not working

### ISSUE THREAD

**Issue by blackearth2014** [mschack/VCV-Rack-Plugins#10]
![mixerbugs](https://user-images.githubusercontent.com/7956901/32863849-e63bdb4c-ca2a-11e7-9666-be7df3162fe0.jpg)

MsHack,

Please observe the screen shot of your updated mixer. Channel D4 and the Master both right meter is not responding to mono audio signals and stereo imputs. As you can see where I have placed the cable as stereo, only the left channel is metering. If I was to connect a mono signal to any of the channels, the master output on the left is the one that meters.

**Comment by blackearth2014** [comment]
The 4x4 console no longer has pre-outs for each section in respect to the added meters. Was this a feature compromise because of space?

**Comment by blackearth2014** [comment]
One more thing. The compressor that is a W.I.P., will it be a feedback compressor or feed forward compressor emulation?

**Comment by mschack** [comment]
Yes I have seen this.  Get the latest version, I think it occurs less often.  Though I don't know the reason it still happens some times.  To get them to work again restart VCV Rack.

**Comment by mschack** [comment]
hey @blackearth2014, I am not sure what the difference is, right now it just attenuates the signal by an amount defined by the ratio knob after the signal goes above the threshold.  Again, I may be ignorant in my design so if you have any suggestions...

**Comment by mschack** [comment]
FYI, I just added a 3 channel cross fader.

**Comment by blackearth2014** [comment]
Here is some basic meanings of an audio compressor:

In a compressor, the DC voltage that controls the VCA level is coming from the AUDIO ITSELF... at some point in the box, the audio signal is split... one goes on it's merry way... the other is mangled into a DC voltage using a rectifier. So now, you can picture it... the audio has been split, one is normal, the other is a DC representation of that same audio.

In **Feedback mode**, the audio will already have gone through the VCA before the audio is split. So it splits (after the VCA), goes through the rectifier, and goes BACK to adjust the next audio signal that is coming in. More musical on voices, basses, guitars and etc.

In **Feedfoward**, the audio is split BEFORE the VCA, so it hasn't been adjusted yet. This means the same audio and DC are hitting at the same time. Use more for limiters and heavy compression.

**Comment by mschack** [comment]
Thanks!.  I was reading a lot on compressors.  My attack and release don't quite work normally, they are very short and restart on every threshold trigger.  I think that is why I put it in the WIP category.  Lots to learn...

**Comment by blackearth2014** [comment]
Do to the fact that most of the modules are synth and drum and voltage control based, I would think that a "Feedforward" module would be easier to learn to program. The release should follow the decay or the release of the audio signal and the decay of a release parameters. Maybe a release hybrid of the Feedback/Feedforward model. Take a look at this vst compressor from Tokyo Dawn Records at http://www.tokyodawn.net/tdr-kotelnikov/. There is a video to give you a idea.

**Comment by mschack** [comment]
Nice.  Might be a while before I achieve anything like that. :)

**Comment by blackearth2014** [comment]
No need to do anything like the Tokyo Dawn compressor. I just wanted to share the information on how they work.

**Comment by mschack** [comment]
Any links to music you have been creating?  I am always interested to see what people can do with this stuff, and not just my modules.

**Comment by blackearth2014** [comment]
I haven't done any full music productions because I am still learning this modular stuff myself. But I will say that your 2 sequencers seems to be more musical based than just twisting knobs and stuff. I will say this though, If you can find a way to implement chord presets based off a set scale, that will be awesome. The Seq Triad can play a c triad or minor base of presets. This will make you mad or crazy, but what if you were to expand the seq triad module to the size of the 4x4 console with 6 piano widgets to produce larger chords for deep jazz house music? If you have a facebook account, I can message you my Rack files of what I have been doing with the modules.

**Comment by blackearth2014** [comment]
Here is a sound cloud link of some of my old scratch compositions and ideas at https://soundcloud.com/blessedhouse.
